kuma/users/views.py

```diff
@@ -179,74 +183,147 @@ def ban_user_and_cleanup_summary(request, user_id):
-        submitted_to_akismet_by_distinct_doc = [rev for rev in submitted_to_akismet if rev.id in distinct_submitted_to_akismet_rev_ids]
-    else:
-        submitted_to_akismet_by_distinct_doc = []
-    if len(not_submitted_to_akismet) > 0:
-        distinct_not_submitted_to_akismet_rev_ids = list(
-            Document.objects
-            .filter(revisions__in=not_submitted_to_akismet)
-            .annotate(latest_rev=Max('revisions'))
-            .values_list('latest_rev', flat=True))
-        not_submitted_to_akismet_by_distinct_doc = [rev for rev in not_submitted_to_akismet if rev.id in distinct_not_submitted_to_akismet_rev_ids]
-    else:
-        not_submitted_to_akismet_by_distinct_doc = []
-    actions_taken = {'revisions_reported_as_spam': submitted_to_akismet_by_distinct_doc}
+    submitted_to_akismet_by_distinct_doc = revision_by_distinct_doc(submitted_to_akismet)
+    not_submitted_to_akismet_by_distinct_doc = revision_by_distinct_doc(not_submitted_to_akismet)
+    revisions_reverted_by_distinct_doc = revision_by_distinct_doc(revisions_reverted_list)
+    revisions_not_reverted_by_distinct_doc = revision_by_distinct_doc(revisions_not_reverted_list)
+    revisions_deleted_by_distinct_doc = revision_by_distinct_doc(revisions_deleted_list)
+    revisions_not_deleted_by_distinct_doc = revision_by_distinct_doc(revisions_not_deleted_list)
+
+    actions_taken = {
+        'revisions_reported_as_spam': submitted_to_akismet_by_distinct_doc,
+        'revisions_reverted_list': revisions_reverted_by_distinct_doc,
+        'revisions_deleted_list': revisions_deleted_by_distinct_doc
+    }
 
     """ The "Needs followup" section """
-    # TODO: Phase III: needs_follow_up will include revisions that have not been reverted
     # TODO: Phase IV: If user made actions while reviewer was banning them
     new_action_by_user = []
     needs_follow_up = {
         'manual_revert': new_action_by_user,
-        'not_submitted_to_akismet': not_submitted_to_akismet_by_distinct_doc
+        'not_submitted_to_akismet': not_submitted_to_akismet_by_distinct_doc,
+        'not_reverted_list': revisions_not_reverted_by_distinct_doc,
+        'not_deleted_list': revisions_not_deleted_by_distinct_doc
     }
 
     """ The "No Actions Taken" section """
     revisions_already_spam = revisions_from_last_three_days.filter(
         id__in=request.POST.getlist('revision-already-spam')
     )
-    distinct_already_spam_rev_ids = list(
-        Document.objects
-        .filter(revisions__in=revisions_already_spam)
-        .annotate(latest_rev=Max('revisions'))
-        .values_list('latest_rev', flat=True))
-    revisions_already_spam_by_distinct_doc = revisions_already_spam.filter(id__in=distinct_already_spam_rev_ids)
+    revisions_already_spam = [rev for rev in revisions_already_spam]
+    revisions_already_spam_by_distinct_doc = revision_by_distinct_doc(revisions_already_spam)
+
+    revisions_already_spam_ids = [rev.id for rev in revisions_already_spam]
```

Comment:
``revisions_already_spam_ids = list(revisions_already_spam.values_list("id", flat=True))`` would work, before the queryset is turned into a list.  However, it doesn't save any database traffic, since you're going to request the full records anyway. I'm OK with the list comprehension here.